### ISSUE TITLE
[GTA SA] SilentPatch wet roads reflections conflict between modloader.asi and wshps.asi

### ISSUE THREAD

**Issue by yugokoral** [thelink2012/modloader#95]
SilentPatch returned the feature of wet roads reflections. But this feature doesn't work, when i use modloader. Can anything be done about that? Thank you

**Comment by thelink2012** [comment]
Make sure you are using the latest version of both Mod Loader and SilentPatch. I've contacted a few people using these mods together and they cannot confirm this issue. Additionally, you might want to try installing SilentPatch outside of Mod Loader (if you did install inside it).

**Comment by yugokoral** [comment]
I tried several times different combinations. Now i have clean instalation, of 1.0 GTA SA US + wshps.asi + widescreenFix.asi + modloader.asi (21/01/2018 - 0.3.7 - 685kB) + SilentPatchSA.asi (22/02/2020 - 276kB).
Modloader and SilentPatch are the latest versions (i hope?) I have found. Everything is in root folder and i didn't put SilentPatch in modloader folder. Also now i have no mods instaled, nor cleo and reflections of car lights on wet roads when raining, still doesn't work. When i remove modloader.asi, it works. So i don't know :(

EDIT: I found another conflict with wshps.asi.  If i use wshps.asi and modloader.asi together, than silentPatch.asi wont show wet road reflestions. If i remove modloader.asi or if i remove wshps.asi, reflections come back. Strange. I am using older version of widescreen fix, so i need wshps.asi, because new wersion of widescreenfix.asi has problems with some textures with speedometers.

**Comment by PRinceBhai** [comment]
I'm using SP and ModL with shit ton of Mods and i still get Wet Roads, have you installed your Copy od San Andreeas outside of Program Files? Also try deleting your gtasa.set which is in the Documents folder.

**Comment by yugokoral** [comment]
> 
> 
> I'm using SP and ModL with shit ton of Mods and i still get Wet Roads, have you installed your Copy od San Andreeas outside of Program Files? Also try deleting your gtasa.set which is in the Documents folder.

Yeah, because you don't use wshps.asi (Widescreen + HOR support).

**Comment by thelink2012** [comment]
So are you saying the problem only occurs when Mod Loader + SilentPatch +
wshps are installed together?

**Comment by yugokoral** [comment]
Yeah. Exactly. NOW I am confused omg. I added SKYGFX.asi mod, and now it WORKS again - together with wshps.asi, modloader.asi, SilentPatch and SKYGFX. Modloader, skygfx and wshps are in root folder. SilentPatch is in scripts folder. 
So i tried every file again and yeah, exactly as i said... Eventualy i will use SKYGFX too. It seems it works alltogether :)

PS: There should be something conflicting in "priorities?" between modloader and wshps, that removes reflections from asphalt roads, what SilentPatch returned to the game. So workaround is to use SKYGFX. 
Maybe someone will need this :) 

EDIT: I don't wanna close the issue, because its just workaround. The problem originally exists anyway. Changed the title, for easier search.

**Comment by thelink2012** [comment]
Glad it works now. I'll leave the issue open for further investigation.

**Comment by PRinceBhai** [comment]
> > 
> > 
> > I'm using SP and ModL with shit ton of Mods and i still get Wet Roads, have you installed your Copy od San Andreeas outside of Program Files? Also try deleting your gtasa.set which is in the Documents folder.
> 
> Yeah, because you don't use wshps.asi (Widescreen + HOR support).

By "shit ton of mods installed" I meant i also have wshps.asi installed.

**Comment by yugokoral** [comment]
> > > I'm using SP and ModL with shit ton of Mods and i still get Wet Roads, have you installed your Copy od San Andreeas outside of Program Files? Also try deleting your gtasa.set which is in the Documents folder.
> > 
> > 
> > Yeah, because you don't use wshps.asi (Widescreen + HOR support).
> 
> By "shit ton of mods installed" I meant i also have wshps.asi installed.

Hmm... do you use SKYGFX also?

**Comment by matiezek** [comment]
same here, i have Skygfx, Silent Patch, modloader and wshps, and if i delete Skygfx, or only wshps, or only modloader i get wet roads back, but with all of them on i don't, i still can't find the issue, i even moved silent patch and wshps to root and it's the same.

**Comment by Julinkk** [comment]
> Oui. Exactement. Là, je suis complètement perdu. J'ai ajouté le mod SKYGFX.asi, et maintenant ça remarche ! Avec wshps.asi, modloader.asi, SilentPatch et SKYGFX. Modloader, SKYGFX et wshps sont à la racine du jeu. SilentPatch est dans le dossier scripts. J'ai donc réessayé chaque fichier et, comme je l'avais dit, ça fonctionne parfaitement. Finalement, j'utiliserai aussi SKYGFX. Apparemment, tout fonctionne ensemble :)
> 
> PS : Il semble y avoir un conflit de priorités entre modloader et wshps, ce qui supprime les reflets sur les routes asphaltées, réintégrés par SilentPatch. La solution consiste donc à utiliser SKYGFX. Cela pourrait être utile à quelqu'un :)
> 
> EDIT : Je ne souhaite pas clore le ticket, car il ne s'agit que d'une solution de contournement. Le problème initial existe bel et bien. J'ai modifié le titre pour faciliter la recherche.



> Yeah. Exactly. NOW I am confused omg. I added SKYGFX.asi mod, and now it WORKS again - together with wshps.asi, modloader.asi, SilentPatch and SKYGFX. Modloader, skygfx and wshps are in root folder. SilentPatch is in scripts folder. So i tried every file again and yeah, exactly as i said... Eventualy i will use SKYGFX too. It seems it works alltogether :)
> 
> PS: There should be something conflicting in "priorities?" between modloader and wshps, that removes reflections from asphalt roads, what SilentPatch returned to the game. So workaround is to use SKYGFX. Maybe someone will need this :)
> 
> EDIT: I don't wanna close the issue, because its just workaround. The problem originally exists anyway. Changed the title, for easier search.

It's been a while, but this is the only place I've found solutions to this problem. Like you, I have wshps.asi, modloader.asi, SilentPatch, and SKYGFX, and it still doesn't work. It only works if I remove wshps, which prevents Widescreen Fix from working, or modloader, which prevents mods from loading. How is your SkyGFX installed?
